From 304ecce39ee5e5d38b4431f07f6a47b0bc87a61e Mon Sep 17 00:00:00 2001
From: Konstantin Makariev <kmakariev@motorola.com>
Date: Mon, 19 Oct 2015 15:56:28 -0500
Subject: [PATCH 383/959] IKSWM-7604 input: synaptics_dsx_i2c: flash
 programming status

Decouple obtaining flash programming status from reading over I2C
bus. Flash programming status becomes known while quering touch IC.
This is stored in in_bootloader variable, which can be used in
synaptics_rmi4_f01_flashprog_show() to report status.

Change-Id: I3a3893ca84548cf354834384177573757e26df2c
Signed-off-by: Konstantin Makariev <kmakariev@motorola.com>
---
 drivers/input/touchscreen/synaptics_dsx_i2c.c | 18 ++----------------
 1 file changed, 2 insertions(+), 16 deletions(-)

diff --git a/drivers/input/touchscreen/synaptics_dsx_i2c.c b/drivers/input/touchscreen/synaptics_dsx_i2c.c
index a1da2ee6605..56ee19d43f7 100644
--- a/drivers/input/touchscreen/synaptics_dsx_i2c.c
+++ b/drivers/input/touchscreen/synaptics_dsx_i2c.c
@@ -2791,23 +2791,9 @@ static ssize_t synaptics_rmi4_irqtimes_show(struct device *dev,
 static ssize_t synaptics_rmi4_f01_flashprog_show(struct device *dev,
 		struct device_attribute *attr, char *buf)
 {
-	int retval;
-	struct synaptics_rmi4_f01_device_status device_status;
 	struct synaptics_rmi4_data *rmi4_data =
-					i2c_get_clientdata(to_i2c_client(dev));
-	retval = synaptics_rmi4_i2c_read(rmi4_data,
-			rmi4_data->f01_data_base_addr,
-			device_status.data,
-			sizeof(device_status.data));
-	if (retval < 0) {
-		dev_err(dev,
-				"%s: Failed to read device status, error = %d\n",
-				__func__, retval);
-		return retval;
-	}
-
-	return scnprintf(buf, PAGE_SIZE, "%u\n",
-			device_status.flash_prog);
+                                        i2c_get_clientdata(to_i2c_client(dev));
+	return scnprintf(buf, PAGE_SIZE, "%d\n", rmi4_data->in_bootloader);
 }
 
 static ssize_t synaptics_rmi4_hw_irqstat_show(struct device *dev,
-- 
2.11.0

